README.md

```diff
@@ -25,84 +19,168 @@
+
+# We execute our query and ask for 10 results only
+results = query.get(10)
+
+# If we want all the results we can iterate over items() 
+# for results in query.items():
+#     print(len(results))
+#     some cool stuff
+
+```
+
+Now that we have our results we can do multiple things, we can iterate over them to get HTTP (or S3) links; we can download the files to a local folder or we can open these files and stream their content directly to other libraries e.g. xarray.
+
+## Accessing the data
+
+### Option 1: we only want the links
+
+If we already have a workflow in place for downloading our data, we can use *earthaccess* as a search-only library and get HTTP links from our query results. This could be the case if our current workflow uses a different language and we only need the links as input.
+
+```python
+
+# if the dataset is cloud hosted there will be S3 links available, the access parameter accepts direct or external, direct access is only possible if we are in us-west-2
+data_links = [granule.data_links(access="direct") for granule in results]
 
 # or if the data is an on-prem dataset
+data_links = [granule.data_links(access="external") for granule in results]
 
-data_links = [granule.data_links(access="onprem") for granule in granules]
+```
 
-# The Store class allows to get the granules from on-prem locations with get()
-# NOTE: Some datasets require users to accept a Licence Agreement before accessing them
-store = Store(auth)
+> Note: as a bonus, *earthaccess* can get S3 credentials for us, or auhenticated HTTP sessions in case we want to use them with a different library.
 
-# This works with both, on-prem or cloud based collections**
-store.get(granules, local_path='./data')
+### Option 2: we want to download the results to a local folder
 
-# if you're in a AWS instance (us-west-2) you can use open() to get a fileset of S3 files!
-fileset = store.open(granules)
+This option is practical if we have the necessary space available on disk, the library will inform us about the approximate size to be downloaded and its progress.
+```python
+for results in query.items():
+    # results is a list of up to 2000 granules (files)
+    earthaccess.download(results,
+                         local_path="./data/atl06/",
+                         auth=auth)
 
-# Given that this is gridded data (Level 3 or up) we could
-xarray.open_mfdataset(fileset, combine='by_coords')
 ```
 
-For more examples see the `Demo` and `EarthdataSearch` notebooks.
+### Option 3: we want to stream the results directly to xarray
 
+This method works better for in-region access when we are working with gridded datasets (processing level 3 and above).
 
-Only **Python 3.8+** is supported.
+```python
+import xarray as xr
+
+ds = xr.open_mfdataset(earthaccess.open(results, auth=auth), engine="scipy")
+
+```
+And that's it! this code works the same for cloud hosted data and we don't have to change it if we are using DAAC hosted datasets.
+
+## More examples:
```

Comment:
Not quite sure what you mean by 'More examples'. Is this list highlighting other capabilities of earthaccess, or examples that you wish to include at a later date in the Readme or readthedocs?